libs/components/core/src/lib/modules/dock/dock.component.ts

```diff
@@ -74,17 +74,16 @@ export class SkyDockComponent {
       );
     }
 
-    const factory = this.#resolver.resolveComponentFactory(component);
     const injector = Injector.create({
       providers: config.providers || [],
       parent: this.#injector,
     });
 
-    const componentRef = this.target.createComponent<T>(
-      factory,
-      undefined,
-      injector
-    );
+    const componentRef = createComponent<T>(component, {
+      environmentInjector: this.#applicationRef.injector,
+      elementInjector: injector,
+    });
+    this.target.insert(componentRef.hostView);
```

Comment:
Any reason we can't use the `target`'s `createComponent` method instead? Seems like a simpler implementation:
```
const componentRef = this.target.createComponent(component, {
  injector,
});
```

Reply:
Good call. That's easier.